server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRDownloadByCounty.java

```diff
@@ -77,21 +73,36 @@ public String endpoint(final Request the_request, final Response the_response) {
       for (final String s : the_request.queryParams()) {
         county_set.add(Integer.valueOf(s));
       }
-      final Set<CastVoteRecord> matches = getMatching(county_set);
-      if (matches == null) {
-        status = HttpStatus.INTERNAL_SERVER_ERROR_500;
-        result = "Error retrieving records from database";
-      } else {
-        try {
-          final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
-          final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
-
-          Main.GSON.toJson(matches, bw);
-          bw.flush();
-        } catch (final IOException e) {
-          status = HttpStatus.INTERNAL_SERVER_ERROR_500;
-          result = "Unable to stream response";
+      try {
+        Persistence.beginTransaction();
+        final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
+        final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+        final JsonWriter jw = new JsonWriter(bw);
+        jw.beginArray();
+        for (final Integer county : county_set) {
```

Comment:
It would be nice to encapsulate the serialization logic so that the endpoint logic proper were clearer. We'll want to do this when we add tests, anyway.